gradio/routes.py

```diff
@@ -971,21 +972,25 @@ async def upload_file(
             try:
                 if upload_id:
                     file_upload_statuses.track(upload_id)
+                max_file_size = app.get_blocks().max_file_size or math.inf
```

Comment:
A bit of an edge case, but this will convert 0 to math.inf

Reply:
Good catch!